test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,60 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Test if Loaning is supported
+    void * msg_pointer{nullptr};
+    rmw_ret_t ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
+    if (ret == RMW_RET_UNSUPPORTED) {
+      rmw_reset_error();
+      GTEST_SKIP();
+    } else {
+      FAIL() << "Tests not ready";
+    }
+    ret = rmw_return_loaned_message_from_publisher(pub, msg_pointer);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    Base::TearDown();
+  }
+};
+
+TEST_F(CLASSNAME(TestPublisherUseLoan, RMW_IMPLEMENTATION), borrow_loaned_message_bad_arguments) {
+  void * msg_pointer = nullptr;
+  rmw_ret_t ret = rmw_borrow_loaned_message(nullptr, ts, &msg_pointer);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  msg_pointer = nullptr;
+  ret = rmw_borrow_loaned_message(pub, nullptr, &msg_pointer);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  msg_pointer = nullptr;
+  ret = rmw_borrow_loaned_message(pub, ts, nullptr);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
```

Comment:
@Lobotuerk why would you expect this to fail?

Reply:
Borrowing on top of an already burrowed msg? IE, pointer not null?

Reply:
That's not clear reading the code, you can add a comment line before the call to the `rmw_borrow_loaned_message`
```suggestion
  // Not null msg_pointer invalid to borrow message
  ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
```

Reply:
Added